opal/mca/hwloc/external/configure.m4

```diff
@@ -146,6 +143,34 @@ AC_DEFUN([MCA_opal_hwloc_external_CONFIG],[
                   AC_CHECK_DECLS([HWLOC_OBJ_OSDEV_COPROC], [], [], [#include <hwloc.h>])
                   AC_CHECK_FUNCS([hwloc_topology_dup])])
 
+           AS_IF([test "$opal_hwloc_external_support" = "yes"],
+                 [AC_MSG_CHECKING([if external hwloc version is 1.5 or greater])
+                  AC_COMPILE_IFELSE(
+                      [AC_LANG_PROGRAM([[#include <hwloc.h>]],
+                          [[
+#if HWLOC_API_VERSION < 0x00010500
+#error "hwloc API version is less than 0x00010500"
+#endif
+                          ]])],
+                      [AC_MSG_RESULT([yes])],
+                      [AC_MSG_RESULT([no])
+                       opal_hwloc_external_support=no])])
+
+           AS_IF([test "$opal_hwloc_external_support" = "yes" &&
+                  test -z "$with_hwloc"],
+                 [AC_MSG_CHECKING([if external hwloc version is 2.0 or greater])
+                  AC_COMPILE_IFELSE(
+                      [AC_LANG_PROGRAM([[#include <hwloc.h>]],
+                          [[
+#if HWLOC_API_VERSION < 0x00020000
+#error "hwloc API version is less than 0x00020000"
+#endif
+                          ]])],
+                      [AC_MSG_RESULT([yes])],
+                      [AC_MSG_RESULT([no])
+                       AC_MSG_WARN([external hwloc version is less than internal version (2.0)])
+                       AC_MSG_WARN([using internal hwloc])
+                       opal_hwloc_external_support=no])])
```

Comment:
I'm not clear on why all this code moved up to the previous block.  The code was previously split into two distinct parts:

1. Do we have external hwloc at all.
2. If we have it, does that external hwloc have all the things we need?

This is not a deal breaker -- it's just odd that you moved *some* of the "does external hwloc have the things we need?" (i.e., the version check) but left other things in the lower block (i.e., the XML check).